clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -107,25 +90,11 @@ public ConfigDef define(String name, Type type, Object defaultValue, Validator v
      * @return This ConfigDef so you can chain calls
      */
     public ConfigDef define(String name, Type type, Object defaultValue, Importance importance, String documentation) {
-        return define(name, type, defaultValue, null, importance, documentation, true);
-    }
-
-    /**
-     * Define a required parameter with no default value
-     *
-     * @param name          The name of the config parameter
-     * @param type          The type of the config
-     * @param validator     A validator to use in checking the correctness of the config
-     * @param importance    The importance of this config: is this something you will likely need to change.
-     * @param documentation The documentation string for the config
-     * @return This ConfigDef so you can chain calls
-     */
-    public ConfigDef define(String name, Type type, Validator validator, Importance importance, String documentation) {
```

Comment:
you removed the constructor that had validator but no default. Was this intentional?